### ISSUE TITLE
Data channel support incompatible with Firefox Nightly

### ISSUE THREAD

**Issue by mquander** [meetecho/janus-gateway#1294]
Firefox now emits m-lines with protocol "UDP/DTLS/SCTP" rather than just "DTLS/SCTP". See [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1335206). This is consistent with recent versions of the spec -- see [here](https://tools.ietf.org/html/draft-ietf-mmusic-sctp-sdp-26). Unfortunately, there are a few places in the Janus source that have hard-coded checks for "DTLS/SCTP", and reject any data channel offers with any other protocol. Those checks should probably be updated.

**Comment by lminiero** [comment]
Yep, I've read some posts by Nils on Facebook and Twitter mentioning this, and I already added an internal memo to address this. Thanks for opening an issue, which will make it easier to track.

**Comment by lminiero** [comment]
> (I'm not sure what to do about "TCP/DTLS/SCTP" since I don't know whether Janus actually supports data channels over TCP. I guess I assumed it did, because all of the UDP-specific stuff that may be in usrsctp seems to be getting abstracted out by us, but I don't actually know.)

No need to worry about that. The profile for media is `UDP/TLS/RTP/SAVPF` as well. This is a known "ambiguity" in SDP, but the actual transport that will be used is not dictated by the profile here, it's agreed upon when exchanging and using candidates.

**Comment by mquander** [comment]
Hmm, something is still going wrong for me in Nightly that didn't happen before and doesn't happen in Chrome -- the symptom is that it looks like ICE never finishes. I will have to figure out whether it's related to this or whether it's just an independent problem.
